libs/designsystem/src/lib/components/modal/services/modal-navigation.service.ts

```diff
@@ -0,0 +1,197 @@
+        : matchedChildRoute;
+      const childSegmentCount = relativeChildRoute.split('/').length;
+      currentRoutePaths.splice(-childSegmentCount); // Pop child segments from end of current route path
+    }
+  }
+
+  private getRoutePaths(routes: Routes, parentPath: string[]): string[] {
+    return Array.isArray(routes)
+      ? ([] as string[]).concat(...routes.map((route) => this.getRoutePath(route, parentPath)))
+      : [];
+  }
+
+  private getRoutePath(route: Route, parentPath: string[]): string[] {
+    const routes: string[] = [];
+    if (!!route.outlet) return routes; // Don't return relative paths for outlet routes
+
+    const currentPath = [...parentPath];
+    if (!!route.path) {
+      currentPath.push(route.path);
+      routes.push(currentPath.join('/'));
+    }
+
+    return routes.concat(this.getRoutePaths(route.children, currentPath));
+  }
+
+  private getModalRoutePath(route: Route, parentPath: string[]): string[] {
+    const modalOutletName = 'modal';
+    if (!!route.path && route.outlet === modalOutletName) {
+      const modalOutletPath = `(${modalOutletName}:${route.path})`;
+      const modalRoutePath = [...parentPath, modalOutletPath].join('/');
+      return [modalRoutePath];
+    }
+    const currentPath = [...parentPath];
+    if (!!route.path) {
+      currentPath.push(route.path);
+    }
+    return ([] as string[]).concat(...this.getModalRoutePaths(route.children, currentPath));
+  }
+
+  private getModalRoutePaths(routes: Routes, parentPath: string[]): string[] {
+    return Array.isArray(routes)
+      ? ([] as string[]).concat(...routes.map((route) => this.getModalRoutePath(route, parentPath)))
+      : [];
+  }
+
+  private isNewModalWindow(navigationEnd: NavigationEnd) {
+    const currentNavigation = this.router.getCurrentNavigation();
+    if (!currentNavigation || !currentNavigation.previousNavigation) {
+      return true;
+    }
+    const previousRoutePath = (
+      currentNavigation.previousNavigation.finalUrl ||
+      currentNavigation.previousNavigation.extractedUrl
+    ).toString();
+    const wasModalRoute = this.isModalRoute(previousRoutePath);
+    const isModalRoute = this.isModalRoute(navigationEnd.urlAfterRedirects);
+    if (!wasModalRoute && isModalRoute) {
+      return true;
+    }
+
+    const currentModalRouteParent = navigationEnd.urlAfterRedirects.split('/(modal:')[0];
+    const previousModalRouteParent = previousRoutePath.split('/(modal:')[0];
+
+    return previousModalRouteParent !== currentModalRouteParent;
+  }
+
+  private navigationEndListener$ = this.router.events.pipe(
+    filter((event): event is NavigationEnd => event instanceof NavigationEnd)
+  );
+
+  modalRouteActivated$ = this.navigationEndListener$.pipe(
+    map((event) => this.isModalRoute(event.urlAfterRedirects)),
+    filter((isModalRoute) => isModalRoute),
+    map(() => this.getCurrentActivatedRoute())
+  );
+
+  modalRouteDeactivatedFor(routeConfig: Routes[]) {
```

Comment:
not sure "modalRouteDeactivedFor" is meaningful to me, at least it requires some context.

Reply:
The `for` hints to the param `routerConfig` - so it reads "when a modal route is deactivated for this given route config" - or at least, thats the idea ;)